spek-ide-plugin/android-studio/src/main/kotlin/org/spekframework/intellij/SpekRunnnerCommandLineState.kt

```diff
@@ -0,0 +1,96 @@
+package org.spekframework.intellij
+
+import com.intellij.execution.DefaultExecutionResult
+import com.intellij.execution.ExecutionResult
+import com.intellij.execution.Executor
+import com.intellij.execution.application.BaseJavaApplicationCommandLineState
+import com.intellij.execution.configurations.JavaParameters
+import com.intellij.execution.configurations.RunConfigurationBase
+import com.intellij.execution.process.ProcessHandler
+import com.intellij.execution.runners.ExecutionEnvironment
+import com.intellij.execution.runners.ProgramRunner
+import com.intellij.execution.testframework.sm.SMTestRunnerConnectionUtil
+import com.intellij.execution.testframework.sm.runner.SMTRunnerConsoleProperties
+import com.intellij.execution.testframework.sm.runner.SMTestLocator
+import com.intellij.execution.ui.ConsoleView
+import com.intellij.execution.util.JavaParametersUtil
+import com.intellij.openapi.extensions.Extensions
+import com.intellij.openapi.roots.OrderEnumerator
+import com.intellij.util.PathUtil
+import org.spekframework.ide.ServiceMessageAdapter
+
+class SpekRunnnerCommandLineState<T>(environment: ExecutionEnvironment, configuration: T)
+    : BaseJavaApplicationCommandLineState<T>(environment, configuration)
+where T: RunConfigurationBase,
+      T: SpekBaseJvmRunConfiguration {
+    override fun createJavaParameters(): JavaParameters {
+        val params = JavaParameters()
+        params.isUseClasspathJar = true
+        val module = myConfiguration.configurationModule
+        val jreHome = if (myConfiguration.isAlternativeJrePathEnabled) {
+            myConfiguration.alternativeJrePath
+        } else {
+            null
+        }
+
+        val pathType = JavaParameters.JDK_AND_CLASSES_AND_TESTS
+        JavaParametersUtil.configureModule(module, params, pathType, jreHome)
+
+        val jars = listOf(
+            PathUtil.getJarPathForClass(ServiceMessageAdapter::class.java)
+        )
+
+        params.classPath.addAll(jars)
+        params.mainClass = "org.spekframework.ide.ConsoleKt"
+
+        // this will set the vm parameters
```

Comment:
nit: comments should be sentences. https://nedbatchelder.com/blog/201401/comments_should_be_sentences.html